sources/web/datalab/server.ts

```diff
@@ -303,13 +303,24 @@ function trimProtocol(requestPath: string) {
   return start > -1 ? requestPath.slice(start + '://'.length) : requestPath;
 }
 
-function socketHandler(request: http.ServerRequest, socket: net.Socket, head: Buffer) {
-  // Websocket requests aren't CORS-checked by the browser. Reject any CORS
-  // requests here by checking their host vs origin request headers.
+function requestIsCrossOrigin(request: http.ServerRequest) {
```

Comment:
The name is inverted from the behavior. This is checking that the request is not cross origin rather than that it is.

I would suggest keeping the name and inverting the behavior (since it would be confusing for the check to be `!requestIsNotCrossOrigin(...)`)

Reply:
Woops, fixing.